index.tt

```diff
@@ -1,157 +1,256 @@
+      Nix makes it <strong>trivial to share development and build
+        environments</strong> for your projects, regardless of what programming
+      languages and tools you’re using.
     </p>
+  </div>
+  <div class="span4">
+    <h2>Reliable</h2>
+    <p>
+      Nix’s ensures that installing or upgrading one package <strong>cannot
+        break other packages</strong>. It allows you to <strong>roll back to
+        previous versions</strong>, and ensures that no package is in an
+      inconsistent state during an upgrade.
+    </p>
+  </div>
+</div>
+
+
+<hr />
 
+<div id="packages-search" class="row-fluid text-center">
+  <div class="span12">
+    <h1>Choose from Thousands of Packages</h1>
+    <p class="text-center lead">
+      The Nix Packages collection (<a href="https://github/NixOS/nixpkgs">nixpkgs</a>) is a set of
+      <strong>over 60 000 packages</strong> for the Nix package manager,
+      released under a permissive <a href="https://spdx.org/licenses/MIT.html">MIT/X11 license</a>.
+    </p>
+    <form>
+      <div class="input-append">
+        <input class="input-xxlarge" placeholder="Search for packages" type="text" />
+        <button class="btn">Search</button>
+      </div>
+      <p class="text-center">or <a href="[% root %]nixos/options.html">search among many NixOS options.</a></p>
+    </form>
   </div>
+</div>
+
+<hr />
+
+<h1 class="text-center">Examples...</h1>
+
+<div id="features">
+
+  <section>
+    <div class="row-fluid">
+      <div class="span6">
+
+        <h2>Try new tools without fear</h2>
+        <p>Don't clutter your system with tools that you use only now and then.</p>
+        <pre class="well" style="margin-top:1em">
+<span class="nix-command">python --version</span>
+python: command not found
+<span class="nix-command"><strong>nix-shell -p python3</strong></span>
+(nix-shell)<span class="nix-command">python --version</span>
+Python 3.7.7</pre>
+
+        <h2>One tool, multiple languages</h2>
+        <p></p>
+        <pre class="well" style="margin-top:1em">
+<span class="nix-command"><strong>nix-shell -p python3 nodejs go rustc</strong></span>
+(nix-shell)<span class="nix-command">node --version</span>
+v10.20.1
+(nix-shell)<span class="nix-command">go version</span>
+go version go1.14.1 linux/amd64
+(nix-shell)<span class="nix-command">rustc --version</span>
+rustc 1.42.0</pre>
+
+      </div>
+      <div class="span6">
 
+        <h2>Isolated development environments</h2>
+        <p>
+          After you get familiar with <code>nix-shell -p</code> you can take
+          the next step further and <a href="[%root%]learn.html">learn some
+            Nix</a>. To setup a more persistent environment you can also write a
+          simple <code>shell.nix</code> file:
+        </p>
+        <pre class="well" style="margin-top:1em">
+{ pkgs ? import &lt;nixpkgs&gt; {}
+}:
+pkgs.mkShell {
+  name = "dev-shell";
+  buildInputs = [
+    pkgs.python3
+    pkgs.python3Packages.virtualenv
```

Comment:
Python developers are already aware of virtualenv or the newer python3 builtin venv and the pain it comes with. I actually find this example more impressive:

```suggestion  
    (pkgs.python3.withPackages (ps: [ps.flask]))
```

because it removes the need for users to keep the virtual environment consistent.

Reply:
But python developers are familiar with virtualenv. Using your example I would have to explain a lot more then this. But I agree with you that your example is much more impressive, but its longer. We can of course link to this tutorial once it is done.